### ISSUE TITLE
Image position in Navigation Page Title View is not consistent when used in Content Page with MasterDetailPage navigation pattern

### ISSUE THREAD

**Issue by pakbaz** [xamarin/Xamarin.Forms#3881]
### Description
Image in Navigation Page Title View Position is not consistent when used in Content Page with MasterDetailPage navigation pattern in new 3.2 version

### Steps to Reproduce

1. adding Custom TitleView for NavigationPage (Forms 3.2 new feature)
2. adding a stack layout with one image added to center for MasterDetailPage style app for different pages
3. run emulator on iOS and android and see inconsistencies

```
 View view = new StackLayout()
            {
                HeightRequest = 40,
                Orientation = StackOrientation.Horizontal,
                HorizontalOptions = LayoutOptions.FillAndExpand
            };
            (view as StackLayout).Children.Add(new Image()
            {
                Aspect = Aspect.AspectFill,
                HorizontalOptions = LayoutOptions.Center
                Source = ImageSource.FromResource("Helix.PassengerPortal.Images.helipass_logo_white.png")
            });
            NavigationPage.SetTitleView(this, view);
```

Tried different combination of properties for image and seen inconsistent result for each.
### Expected Behavior
having one image centered on Navigation Page Title in the center of the page
### Actual Behavior
on android image is not centered, on iOS navigating from page to page, initially centered, then not centered, and sometime not entirely visible
### Basic Information

- Version with issue: 3.2
- IDE: Visual Studio on Mac
- Platform Target Frameworks: 
  - iOS:  12
  - Android: 8
- Affected Devices: iOS, android

**Comment by PureWeen** [comment]
@pakbaz   two things on this one
1) This will be fixed by this PR https://github.com/xamarin/Xamarin.Forms/issues/3679
2) Horizontal stacklayouts and those HorizontalOptions behaviors are a little bit confusing. Creating the Stacklayout like you have it there won't center the image in a stacklayout.  If you move that StackLayout to just the content of a ContentPage you'll see it also doesn't get centered (look below).  The image on iOS was initially centered because the entire view is being centered in the container but the image inside the view isn't technically centered

![image](https://user-images.githubusercontent.com/5375137/46235750-05a43580-c339-11e8-8eae-2eae7299928c.png)

**Comment by samhouts** [comment]
For additional clarity, if you need the image to be centered in this case, use "CenterAndExpand". Thanks!

**Comment by PureWeen** [comment]
The fix for this has been pushed to the latest 3.3.0 SR release.
Please let us know if that's resolved your issue
